internal/event.go

```diff
@@ -0,0 +1,51 @@
+/**
+ *  Copyright (c) 2020  Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package internal
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+)
+
+type (
+	// An Event bundles even payload and metadata that travels with the event
+	Event struct {
+		ctx      context.Context
+		Payload  interface{} `json:"payload"`  // event payload (could also be string, but preparsed i sprobably more efficient if we want to allow deep inspection)
+		Encoding string      `json:"encoding"` // optional encoding hint to be set by input plugin
+		Metadata interface{} `json:"metadata"` // optional metadata produced by filter chain
+		AckTree  AckTree     `json:"ackTree"`  // optional ack chain (or ack tree)
```

Comment:
Please reference the ackTree package under internal/pkg/ack

Reply:
don't think i can do this until i merge, but will do it then